### ISSUE TITLE
[BUG] Multiple interactions do not collapse into a single notification

### ISSUE THREAD

**Issue by aurynn** [mastodon/mastodon-ios#399]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Receiving notifications of interactions (Favourites or Boosts) currently generate a new notification per interaction. This results in a huge number of notifications at the phone level (a badge count in the hundreds), as well as a huge number of notifications to scroll through in the Notifications timeline.

### Expected Behavior

Similar to the Twitter iOS app, it would be great if interactions on the same Toot would collapse into a single notification, to reduce the visual noise and alert fatigue that results from a Toot receiving a significant number of Favourites or Boosts.

This could optionally be a setting.

### Steps To Reproduce

1. Post a Toot.
2. Toot starts trending
3. Observe large number of notifications on the device.

### Environment

```markdown
- Device: iPhone XS
- OS: 15.4.1
- Version: v1.3.3
- Build: 116
```

**Comment by kemayo** [comment]
Agreed that this is the major thing making me want to use a different client. (E.g. Ice Cubes bundles notifications.)

**Comment by delusho** [comment]
Agreed! It would be truly great to have that !!

**Comment by brendanjones** [comment]
Not a bug, but definitely a badly-needed feature. In my eyes this should be number one priority. 

The iOS app has come a long way since I checked it out a couple of years ago and promptly uninstalled it in favour of 3rd party apps like Mona. It’s now looking great, and the app not grouping notifications is one of the few remaining things keeping me away. This and not being able to hide replies and boosts from my feed. 

I recently had a post get 2000+ boosts and favs - can you imagine how my notifications looked without grouping? 😅
